deep_quoridor/src/utils/timer.py

```diff
@@ -0,0 +1,83 @@
+import time
+from typing import Optional
+
+import wandb
+import wandb.wandb_run
+
+from utils.misc import format_time
+
+
+class Timer:
+    """
+    Class to measure elapsed time and log it to the console and optionally wandb.
+    It's a class just to create a container for the functions and variables, it can't be instantiated.
+    TO DO: would be nice if it would easily support multi-process, so we can gather information coming
+    from different processes and compile it.
+    """
+
+    starts = {}
+    counts = {}
+    totals = {}
+    wandb_run: Optional[wandb.wandb_run.Run] = None
```

Comment:
Since wandb_run is a class variable and not an instance variable, this means that if you create multiple timer instances and set the wandb_run for one of them, they all will use that, right? (this seems like a good thing)

Reply:
You can't create instances of this class, the method below won't let you.  The idea is just to use the static methods.  I could have created just normal functions, but I liked this as a way of keeping everything in the same "namespace"